src/hooks/useWallet.ts

```diff
@@ -48,7 +48,7 @@ export default function useWallet() {
             accounts: getAccountsByProvider(id),
             isActive: activeAccount?.providerId === id,
             isConnected: connectedAccounts.some((accounts) => accounts.providerId === id),
-            connect: () => connect(id),
+            connect: (onDisconnect, email) => connect(id, onDisconnect, email),
```

Comment:
```suggestion
            connect: (email?: string) => connect(id, email),
```

There are a lot of functions named `connect` in the library, so it's easy to confuse them, especially here in the useWallet hook where all 3 appear.

1. This (highlighted) is the `provider.connect` function exposed to the consuming app in useWallet's `providers` array. Previously it didn't accept any arguments but now it will take an optional `email` argument for Magic Link.
2. The useWallet hook has an internal `connect` function, which is being called here. It accepts a provider ID and finds the matching wallet client and calls its `connect` method. It'll now also accept the optional `email` arg.
3. The BaseClient `connect` method is the one with the `onDisconnect` callback. That's how it's passed the `clearAccounts` function, which updates the Zustand store when the provider is disconnected.

Just typing all that out makes me realize how confusing it is! Renaming the hook's internal `connect` function to something like `connectProvider` would help differentiate them a bit. 😅

Suffice it to say, you don't need to add the `onDisconnect` argument here in the `connect` function for the public API.

This change will trigger a few other subsequent changes... I'll make them in additional comments (without the long explanation).